### PR TITLE
Implemented Sheet class, added PointOfInterest stub

### PR DESCRIPTION
**Pull request by nolmoonen**
#### Deployment instructions
None.

#### What have I done
Implemented Sheet class and unit tests for Sheet class. Added stub for PointOfInterest since Sheet depends on it.

#### Testing
Run Sheet unit tests.

### CODE REVIEW

**Comment by nolmoonen** [comment]
Could someone explain to me why the checks have failed?

**Comment by Dacaspex** [comment]
If a build fails, you can inspect the build logs on Travis. The build on Travis works as follows:
- First, compile the application
- Secondly, run the tests on Firefox and Chrome. You can do this yourself too using the `ng test --watch=false` from your command line
- Lastly, it runs a linter to check your code style. You can do that too using `ng lint` in your command line

If any check fails, the build fails and you get red flags. Let's see what happened here:
image
As we can see, the tests ran fine and everything looks good! (Exit code 0 means everything is fine)

Let's take a look at the linter
image
Oh oh, something went wrong while linting. You can always run the `ng lint` command in your command line too see the output in your console. I can see two things that went wrong, you used `"` instead of `'` and you used the typehint `number`. We have defined to only use `'` for consistency and also agreed that types that can be derived from their value are not needed. 

I think that this should help you fix the issues! :)

File: src/app/backend/models/sheet/sheet.spec.ts [hunk 1]
```diff
@@ -1,7 +1,92 @@
-import { Sheet } from './sheet';
+import {Sheet} from './sheet';
+import {TimeService} from "../../time/time.service";
+import {PointOfInterest} from "../pointofinterest/pointOfInterest";
```
Comment (Dacaspex) [inline src/app/backend/models/sheet/sheet.spec.ts]: Did you import the codestyle.xml from the Drive? Imports should have a whitespace between the opening and closing curly braces, like so:
```javascript
import { TimeService } from "..."
```

File: src/app/backend/models/sheet/sheet.ts [hunk 2]
```diff
@@ -1,2 +1,148 @@
...
+
+    /**
+     * Time this sheet starts on the path.
+     */
+    private scheduledMoment: number;
+
+    /**
+     * Last point of interest.
+     */
+    private lastPointOfInterest: PointOfInterest;
```
Comment (Dacaspex) [inline src/app/backend/models/sheet/sheet.ts]: Just wondering, should it also have the next point of interest?

File: src/app/backend/models/sheet/sheet.ts [hunk 3]
```diff
@@ -1,2 +1,148 @@
+import {TimeService} from "../../time/time.service";
+import {PointOfInterest} from "../pointofinterest/pointOfInterest";
```
Comment (Dacaspex) [inline src/app/backend/models/sheet/sheet.ts]: Code styling just like the other imports

File: src/app/backend/models/sheet/sheet.ts [hunk 4]
```diff
@@ -1,2 +1,148 @@
...
+     */
+    private scheduledMoment: number;
+
+    /**
+     * Last point of interest.
+     */
+    private lastPointOfInterest: PointOfInterest;
+
+    constructor(private timeService: TimeService) {
+        // inject TimeService dependency
```
Comment (Dacaspex) [inline src/app/backend/models/sheet/sheet.ts]: No `this.timeService = timeService`? I'm confused?

Comment (nolmoonen) [reply src/app/backend/models/sheet/sheet.ts]: I have been informed this is not necessary, nor does it seem that way from testing.

File: src/app/backend/models/sheet/sheet.ts (hunk 3 above)
Comment (nolmoonen) [reply src/app/backend/models/sheet/sheet.ts]: See other comment.

File: src/app/backend/models/sheet/sheet.ts (hunk 2 above)
Comment (nolmoonen) [reply src/app/backend/models/sheet/sheet.ts]: As far as I have been able to find in the requirements, it should not.

File: src/app/backend/models/sheet/sheet.spec.ts (hunk 1 above)
Comment (nolmoonen) [reply src/app/backend/models/sheet/sheet.spec.ts]: No I did not, will do and will fix this.

File: src/app/backend/models/sheet/sheet.ts (hunk 4 above)
Comment (Dacaspex) [reply src/app/backend/models/sheet/sheet.ts]: JavaScript truly is a miracle

File: src/app/backend/models/sheet/sheet.ts (hunk 2 above)
Comment (Dacaspex) [reply src/app/backend/models/sheet/sheet.ts]: 👍

File: src/app/backend/models/sheet/sheet.spec.ts (hunk 1 above)
Comment (nolmoonen) [reply src/app/backend/models/sheet/sheet.spec.ts]: @Dacaspex please add instructions how to import code style to `READE.md`, as well as how to install dependencies. (however, I managed for now)

**Pull request merged**